src/hx/files/File.hx

```diff
@@ -479,7 +479,16 @@ class File {
          return;
 
       #if (sys || macro || nodejs)
+         #if (flash || openfl)
+         var dir:String = haxe.io.Path.directory(path.toString());
```

Comment:
Thanks for your contribution! could you add a short inline comment describing the reason for this extra code block?

Reply:
Sure. Just a moment.